packages/wrangler/src/__tests__/dev2.test.tsx

```diff
@@ -0,0 +1,153 @@
+      entry: {
+        file: "./index.js",
+        directory: process.cwd(),
+        format: "modules",
+      },
+    });
+    expect(testRenderer.toJSON()).toMatchInlineSnapshot(`
+      <ink-box
+        style={
+          Object {
+            "borderStyle": "round",
+            "flexDirection": "row",
+            "flexGrow": 0,
+            "flexShrink": 1,
+            "marginBottom": 0,
+            "marginLeft": 0,
+            "marginRight": 0,
+            "marginTop": 0,
+            "paddingBottom": 0,
+            "paddingLeft": 1,
+            "paddingRight": 1,
+            "paddingTop": 0,
+          }
+        }
+      >
+        <ink-text
+          internal_transform={[Function]}
+          style={
+            Object {
+              "flexDirection": "row",
+              "flexGrow": 0,
+              "flexShrink": 1,
+              "textWrap": "wrap",
+            }
+          }
+        >
+          B to open a browser, D to open Devtools, L to turn off local mode, X to exit
+        </ink-text>
+      </ink-box>
+    `);
+    await TestRenderer.act(async () => {
+      // TODO: get rid of these sleep statements
+      await sleep(50);
+      testRenderer.unmount();
+      await sleep(50);
+    });
+    await sleep(50);
+  });
+});
+
+async function renderDev({
+  name,
+  entry = { file: "index.js", directory: "", format: "modules" },
+  port,
+  inspectorPort = 9229,
+  accountId,
+  legacyEnv = true,
+  initialMode = "local",
+  jsxFactory,
+  jsxFragment,
+  localProtocol = "http",
+  upstreamProtocol = "https",
+  rules = [],
+  bindings = {
+    kv_namespaces: [],
+    vars: {},
+    durable_objects: { bindings: [] },
+    r2_buckets: [],
+    wasm_modules: undefined,
+    text_blobs: undefined,
+    unsafe: [],
+  },
+  public: publicDir,
+  assetPaths,
+  compatibilityDate,
+  compatibilityFlags,
+  usageModel,
+  buildCommand = {},
+  enableLocalPersistence = false,
+  env,
+  zone,
+}: Partial<DevProps>): Promise<TestRenderer.ReactTestRenderer> {
+  let instance: TestRenderer.ReactTestRenderer;
+  await TestRenderer.act(async () => {
+    instance = TestRenderer.create(
+      <Dev
+        name={name}
+        entry={entry}
+        env={env}
+        rules={rules}
+        port={port}
+        inspectorPort={inspectorPort}
+        legacyEnv={legacyEnv}
+        buildCommand={buildCommand}
+        initialMode={initialMode}
+        localProtocol={localProtocol}
+        upstreamProtocol={upstreamProtocol}
+        jsxFactory={jsxFactory}
+        jsxFragment={jsxFragment}
+        accountId={accountId}
+        assetPaths={assetPaths}
+        public={publicDir}
+        compatibilityDate={compatibilityDate}
+        compatibilityFlags={compatibilityFlags}
+        usageModel={usageModel}
+        bindings={bindings}
+        enableLocalPersistence={enableLocalPersistence}
+        zone={zone}
+      />
+    );
+  });
+  // @ts-expect-error How do I avoid the "used before assigned" error?
```

Comment:
I believe you just need to specify the type as undefinable...

```ts
let instance: TestRenderer.ReactTestRenderer|undefined;
```

Reply:
sweet, that works!